lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -365,10 +365,12 @@ class DeadFunctionElimination : FunctionLivenessComputation {
   /// Removes all dead methods from vtables and witness tables.
   void removeDeadEntriesFromTables() {
     for (SILVTable &vTable : Module->getVTableList()) {
-      vTable.removeEntries_if([this](SILVTable::Pair &entry) -> bool {
+      vTable.removeEntries_if([&](SILVTable::Pair &entry) -> bool {
         if (!isAlive(entry.second)) {
           DEBUG(llvm::dbgs() << "  erase dead vtable method " <<
                 entry.second->getName() << "\n");
+          // Update cache:
+          Module->removeFromVTableCache(&vTable, entry.first);
```

Comment:
SILVTable doesn't have the module as a class member and passing the module to remove_if would be a problem: the function is templated (typename being the Predicate), implemented inside the header file, the header file has a forward deceleration of the SIL Module (can't include it because SIL Module uses SILVTable)